internal/cli/cli.go

```diff
@@ -1566,3 +1571,16 @@ func installedStatus(installed bool) string {
 	}
 	return "missing"
 }
+
+func unsetAsdfReservedEnvVars() error {
+	// These are environment variables which are passed via env or exec.
+	// We strip these out to avoid any potential issues with recursive calls to asdf.
+	asdfManagedVars := []string{"ASDF_INSTALL_TYPE", "ASDF_INSTALL_VERSION", "ASDF_INSTALL_PATH"}
+	for _, v := range asdfManagedVars {
+		err := os.Unsetenv(v)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
```

Comment:
It might be good to write tests that validate this is doing what we expect it to. The only way I can think to test this would be by testing the `asdf env` command, which normally passes the env through unchanged.

Reply:
Added such a test. Confirmed it fails without the change. Thanks for the sample to work off of!